app/src/main/java/com/pokemonshowdown/data/ItemDex.java

```diff
@@ -3,83 +3,47 @@
+import java.util.Arrays;
+import java.util.List;
 
 public class ItemDex {
     private final static String ITAG = ItemDex.class.getName();
-    private static ItemDex sItemDex;
-    private HashMap<String, String> mItemDexEntries;
+    public final static JSONObject DUMMY_OBJECT;
 
-    private ItemDex(Context appContext) {
-        mItemDexEntries = readFile(appContext);
-    }
-
-    private HashMap<String, String> readFile(Context appContext) {
-        HashMap<String, String> ItemDexEntries = new HashMap<>();
-        String jsonString;
+    static {
         try {
-            InputStream inputStream = appContext.getResources().openRawResource(R.raw.item);
-            BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(inputStream));
-            StringBuilder stringBuilder = new StringBuilder();
-
-            String line;
-            while ((line = bufferedReader.readLine()) != null) {
-                stringBuilder.append(line);
-                stringBuilder.append("\n");
-            }
-            jsonString = stringBuilder.toString();
-            inputStream.close();
-
-            JSONObject jsonObject = new JSONObject(jsonString);
-
-            Iterator<String> keys = jsonObject.keys();
-
-            while (keys.hasNext()) {
-                String key = keys.next();
-                JSONObject entry = jsonObject.getJSONObject(key);
-                ItemDexEntries.put(key, entry.toString());
-            }
+            DUMMY_OBJECT = new JSONObject("{\"gen\":1,\"spritenum\":0,\"num\":0,\"name\":\"\",\"id\":\"dummy\",\"desc\":\"\"}");
         } catch (JSONException e) {
-            Log.d(ITAG, "JSON Exception");
-        } catch (IOException e) {
-            Log.d(ITAG, "Input Output problem");
-        }
-
-        return ItemDexEntries;
-    }
-
-    public static ItemDex get(Context c) {
-        if (sItemDex == null) {
-            sItemDex = new ItemDex(c.getApplicationContext());
+            // Will not happen, but is needed because it's final.
+            throw new Error();
         }
-        return sItemDex;
     }
 
     public static int getItemIcon(Context appContext, String itemName) {
+        itemName = itemName.replaceFirst("itemdex_", "");
         return appContext.getResources()
                 .getIdentifier("item_" + MyApplication.toId(itemName), "drawable", appContext.getPackageName());
     }
 
-    public HashMap<String, String> getItemDexEntries() {
-        return mItemDexEntries;
+    public static List<String> getItemDexEntries(Context context) {
+        return Arrays.asList(context.getResources().getStringArray(R.array.itemdex_ids));
     }
 
-    public String getItemName(String name) {
+    public static String getItemName(Context context, String name) {
         name = MyApplication.toId(name);
         String item;
         try {
-            item = mItemDexEntries.get(name);
+            name = name.startsWith("itemDex_") ? name : "itemdex_" + name;
```

Comment:
Arg ok sorry. Thanks for listening to my suggestions and lemme know via comment when it's ready for review